handshake/handshake_message.go

```diff
@@ -12,27 +12,36 @@ import (
 	"github.com/lucas-clemente/quic-go/utils"
 )
 
+// A HandshakeMessage is a handshake message
+type HandshakeMessage struct {
+	Tag  Tag
+	Data map[Tag][]byte
+}
+
+var _ fmt.Stringer = &HandshakeMessage{}
+
 // ParseHandshakeMessage reads a crypto message
-func ParseHandshakeMessage(r io.Reader) (Tag, map[Tag][]byte, error) {
+func ParseHandshakeMessage(r io.Reader) (hm HandshakeMessage, err error) {
```

Comment:
I'm not a fan of having named return values for longer functions (and, in particular, mixing naked and normal returns). Unless you have a very specific reason for using them here, I'd strongly prefer using normal returns.